torch/onnx/_internal/fx/export.py

```diff
@@ -78,62 +189,37 @@ def _create_onnx_friendly_decomposition_table() -> Dict[
         # 2. fx_node_arg (variable in torch.fx.Graph) is be mapped to
-        #    torch.jit.Value, fx_name_to_ts_value[fx_node_arg.name],
+        #    torch.jit.Value, fx_name_to_onnxscipt_tensor[fx_node_arg.name],
         #    in TorchScript graph.
-        ts_value = fx_name_to_ts_value[ts_value.name]
-    elif isinstance(ts_value, float):
-        # Always promote scalar to tensor with element type "dtype."
-        # Usually, "dtype" is extracted from the expected output tensor of the node.
-        # If this assumption is broken, we probably need to
-        #  1. add "scalar" type in ONNX  and extend all exporters to support it, or
-        #  2. write type promotion logic for each operator.
-        # TODO(wechi): the called exporting function should tell all allowed input and output types.
-        # Then, here we can try type-casting if type-mismatch happens.
-        ts_value = g.op("Constant", value_t=torch.tensor(ts_value, dtype=torch.float))
-    elif isinstance(ts_value, int):
-        # Always promote scalar to tensor with element type "dtype."
-        # Usually, "dtype" is extracted from the expected output tensor of the node.
-        # If this assumption is broken, we probably need to
-        #  1. add "scalar" type in ONNX  and extend all exporters to support it, or
-        #  2. write type promotion logic for each operator.
-        # TODO(wechi): the called exporting function should tell all allowed input and output types.
-        # Then, here we can try type-casting if type-mismatch happens.
-        ts_value = g.op("Constant", value_t=torch.tensor(ts_value, dtype=torch.int64))
-    elif ts_value is None:
-        ts_value = g.op("prim::Constant")
-        ts_value.setType(torch._C.OptionalType.ofTensor())
-    elif isinstance(ts_value, list) and all(isinstance(val, int) for val in ts_value):
-        ts_value = g.op("Constant", value_t=torch.tensor(ts_value, dtype=torch.int64))
-    elif isinstance(ts_value, list) and all(isinstance(val, float) for val in ts_value):
-        ts_value = g.op("Constant", value_t=torch.tensor(ts_value, dtype=torch.float))
-    elif isinstance(ts_value, torch.dtype):
-        from torch.onnx import _type_utils
-
-        ts_value = _type_utils.JitScalarType.from_dtype(ts_value)
-    else:
-        raise RuntimeError(f"Unexpected type of fx_node_arg: {type(fx_node_arg)}")
-    return ts_value
+        onnx_tensor = fx_name_to_onnxscipt_tensor[onnx_tensor.name]
+    elif isinstance(onnx_tensor, torch.dtype):
+        onnx_tensor = _type_utils.JitScalarType.from_dtype(onnx_tensor)
+    # else:
+    #     raise RuntimeError(f"Unexpected type of fx_node_arg: {type(fx_node_arg)}")
+    return onnx_tensor
 
 
-def _wrap_fx_args_as_ts_args(g, root, node, fx_name_to_ts_value):
+def _wrap_fx_args_as_ts_args(root, node, fx_name_to_onnxscipt_tensor):
     """Map all FX arguments of a node to arguments in TorchScript graph."""
 
+    del root  # unused
```

Comment:
Don't do that. You never know if this used outside. If you really want to eliminate it, change
```
def _wrap_fx_args_as_ts_args(root, node, fx_name_to_onnxscipt_tensor)
```

to
```
def _wrap_fx_args_as_ts_args(node, fx_name_to_onnxscipt_tensor)
```

Reply:
I don't think this will cause anything on the outside? Seen in https://google.github.io/styleguide/pyguide.html#214-decision

Reply:
Actually you are right. We don't need root. I kept the signature unchanged because I thought it was some fx convention.